src/msg/async/Event.cc

```diff
@@ -260,8 +247,7 @@ void EventCenter::delete_time_event(uint64_t id)
   if (id >= time_event_next_id)
     return ;
 
-  for (map<utime_t, list<TimeEvent> >::iterator it = time_events.begin();
-       it != time_events.end(); ++it) {
+  for (auto it = time_events.begin(); it != time_events.end(); ++it) {
```

Comment:
better off using `auto&` to avoid the copying. might want to consider using the range-for statement instead, see http://www.stroustrup.com/C++11FAQ.html#for.